### ISSUE TITLE
Propose a config option for validation-name default strategy?

### ISSUE THREAD

**Issue by mashpie** [wearebraid/vue-formulate#291]
The current default strategy to use `name` as the default `${name}` in validation-messages is focused more technically than informational for a user. IMHO choosing `label` as the default would resolve much more use cases, while still allowing to customize `validationName` directly

Example:

```vue
<FormulateForm method="POST" v-model="account" @submit="handleSubmit">
    <FormulateInput
      label="E-Mail"
      validationName="E-Mail"
      validation="bail|required|email"
      type="email"
      name="mailrcpt"
      autocomplete="email"
    />
  </FormulateForm>
```

Having a global option or hook to configure that `name` or switch strategy to use `label` as the default would reduce boilerplate.

Thanks for your awesome work!

**Comment by justin-schroeder** [comment]
I think this generally makes sense. You can configure this with a pretty simple plugin, but some kind of global strategy would be nice too. In the meantime, try this plugin (or something like it):

```js
function useLabelAsValidationName (instance) {
  instance.parentValidationMessage = instance.validationMessage
  instance.validationMessage = function (rule, validationContext, vm) {
    validationContext.name = vm.label
    return instance.parentValidationMessage(rule, validationContext, vm)
  }
}

// Then just register your plugin when you initialize
Vue.use(VueFormulate, {
  plugins: [useLabelAsValidationName]
})
```

**Comment by mashpie** [comment]
Charming solution :)

Thanks, @justin-schroeder, for pointing. I am looking forward to 2.5.x while using a plugin for now.

Should we close this FR or leave open for reference?

**Comment by justin-schroeder** [comment]
Let’s keep it open so i can track the “Strategy” feature

**Comment by justin-schroeder** [comment]
@mashpie good news, this is now implemented in the `release/2.5.0` branch. It allows you to define a new gloabl configuration option `validationNameStrategy` to define your own priority sequence as an array:

```js
Vue.use(VueFormualte, {
  validationNameStrategy: ['validationName', 'name', 'label', 'type']
})
```

It also supports using a function which is passed the full instance of the component. Early docs available here:

https://vueformulatecom-git-release-250.braid.vercel.app/guide/validation/#validation-name-strategy
